src/main/scala/com/gu/CheckPriceRisePreConditions.scala

```diff
@@ -0,0 +1,54 @@
+package com.gu
+
+import com.gu.FileImporter.PriceRise
+
+trait PriceRisePreCondition
+case object SubscriptionIsAutoRenewable extends PriceRisePreCondition
+case object SubscriptionIsActive extends PriceRisePreCondition
+case object DeliveryRegionMatchesCurrency extends PriceRisePreCondition
+case object PriceRiseDateIsOnInvoicedPeriodEndDate extends PriceRisePreCondition
+case object ImportHasCorrectCurrentPrice extends PriceRisePreCondition
+case object TargetPriceRiseIsNotMoreThanTheCap extends PriceRisePreCondition
+case object TargetPriceRiseIsNotMoreThanDefaultProductRatePlanChargePrice extends PriceRisePreCondition
+case object TargetPriceRiseIsMoreThanTheCurrentPrice extends PriceRisePreCondition
+case object ThereDoesNotExistAFutureAmendmentOnThePriceRiseDate extends PriceRisePreCondition
+case object CurrentlyActiveProductRatePlanIsGuardianWeeklyRatePlan extends PriceRisePreCondition
+case object TargetPriceRiseIsNotLessThanCataloguePriceWhenRemovingAllDiscountRatePlans extends PriceRisePreCondition
```

Comment:
Not less than the target or current catalog price?

FYI - There may be overrides on the old price adjusting it to be lower than the old catalog price, irrespective of any discount plans.